### ISSUE TITLE
segmentation fault while deserializing sandbox

### ISSUE THREAD

**Issue by zachrammell** [digipen-team-uwu/uwu-engine#4]
https://github.com/digipen-team-uwu/uwu-engine/blob/c1b9b24b3c1a6359e4c28b60e3ff9b8553055304/src/Debugs/TraceLogger.cpp#L75
Call stack:
```
(anonymous namespace)::string_format(const char *, typedef __va_list_tag __va_list_tag *) TraceLogger.cpp:75
TraceLogger::Assert TraceLogger.cpp:214
DeserializeLevel Serialization.cpp:200
GameStateMachine::Enter_BrayanSBOX BrayanSBOX.cpp:34
GameStateMachine::Enter GameStatesManager.h:17
StateMachineUpdater::StartUp StateMachine.cpp:9
Engine::StartUp Engine.cpp:24
main main.cpp:9
```
This code likely only works correctly under MSVC.
Crashes under Debian Bullseye, gcc 9.2.1

**Comment by zachrammell** [comment]
To be fixed by 5558bede7551d3c8f9b1be1fa7a7a98466d89c72 once tested.

**Comment by zachrammell** [comment]
fixed by 5558bede7551d3c8f9b1be1fa7a7a98466d89c72
